src/backend/utils/fmgr/fmgr.c

```diff
@@ -832,7 +835,15 @@ fmgr_security_definer(PG_FUNCTION_ARGS)
 		fcinfo->flinfo = &fcache->flinfo;
 
 		/* See notes in fmgr_info_cxt_security */
-		pgstat_init_function_usage(fcinfo, &fcusage);
+
+		pgstat_init_function_usage_wrapper(fcinfo, &fcusage, cacheTupleProcname);
```

Comment:
if the hook in earlier comment is not null and the dialect is TSQL then we call this func using hook otherwise we will fall back to pgstat_init_function_usage